### ISSUE TITLE
Automatic testing

### ISSUE THREAD

**Issue by luxifer** [Azure/azure-storage-php#132]
Are there any plan to setup automated tests with a CI like appveyor?

**Comment by XiaoningLiu** [comment]
@luxifer Yes, we have planned. The only problem is providing mocking test against CI system. Current test cases run against real storage account for live testing. And we cannot save connection string on CI.

**Comment by luxifer** [comment]
I understand your concern. I tried to run the tests locally with the emulated storage but some parts of the storage are not implemented so the tests did not pass. Some CI provides features to configure an env variable that is not visible publicly but accessible inside the pipeline like https://www.appveyor.com/docs/build-configuration/#secure-variables

**Comment by XiaoningLiu** [comment]
@luxifer Much appreciated for your suggestions!! We did investigation for CIs like Travis-CI. Although we can hide the secret from the logs, it sill has risk, because everyone can submit a PR with some "scripts" to dump up the security string. We have discussed about the risk for that approach. By all means, our next plan to is to achieve mock testing for UT/FT validation.
